core/t_set.go

```diff
@@ -0,0 +1,290 @@
+	}
+	key := args[0]
+
+	// Get the set object from the store.
+	obj := Get(key)
+
+	var count int = 0
+	if obj == nil {
+		return Encode(count, false)
+	}
+	if obj.TypeEncoding != ObjTypeSet|ObjEncodingHT {
+		return Encode(errors.New("WRONGTYPE Operation against a key holding the wrong kind of value"), false)
+	}
+
+	// Get the set object.
+	set := obj.Value.(map[string]bool)
+	for i := 1; i < len(args); i++ {
+		if set[args[i]] {
+			delete(set, args[i])
+			count++
+		}
+	}
+	return Encode(count, false)
+}
+
+func evalSCARD(args []string) []byte {
+	if len(args) != 1 {
+		return Encode(errors.New("ERR wrong number of arguments for 'SCARD' command"), false)
+	}
+
+	key := args[0]
+
+	// Get the set object from the store.
+	obj := Get(key)
+
+	if obj == nil {
+		return Encode(0, false)
+	}
+
+	// If the object exists, check if it is a set object.
+	if obj.TypeEncoding != ObjTypeSet|ObjEncodingHT {
+		return Encode(errors.New("WRONGTYPE Operation against a key holding the wrong kind of value"), false)
+	}
+
+	// Get the set object.
+	set := obj.Value.(map[string]bool)
+	var count int = 0
+	for k, flag := range set {
+		if !flag {
+			delete(set, k)
+		} else {
+			count++
+		}
+	}
+	return Encode(count, false)
+}
+
+func evalSDIFF(args []string) []byte {
+	if len(args) < 1 {
+		return Encode(errors.New("ERR wrong number of arguments for 'SDIFF' command"), false)
+	}
+
+	srcKey := args[0]
+	obj := Get(srcKey)
+
+	srcSet := make(map[string]bool)
+
+	// Get the set object from the store.
+	// store the count as the number of elements in the first set
+	// we decrement the count as we find the elements in the other sets
+	// if the count is 0, we skip further sets but still get them from
+	// the store to check if they are set objects and update their last accessed time
+
+	var count int = 0
+	if obj != nil {
+		if obj.TypeEncoding != ObjTypeSet|ObjEncodingHT {
+			return Encode(errors.New("WRONGTYPE Operation against a key holding the wrong kind of value"), false)
+		}
+		// create a deep copy of the set object
+		srcSet = make(map[string]bool)
+		for k, flag := range obj.Value.(map[string]bool) {
+			if flag {
+				srcSet[k] = flag
+				count++
+			}
+		}
+	}
+
+	for i := 1; i < len(args); i++ {
+		// Get the set object from the store.
+		obj := Get(args[i])
+
+		if obj == nil {
+			continue
+		}
+
+		// If the object exists, check if it is a set object.
+		if obj.TypeEncoding != ObjTypeSet|ObjEncodingHT {
+			return Encode(errors.New("WRONGTYPE Operation against a key holding the wrong kind of value"), false)
+		}
+
+		// only if the count is greater than 0, we need to check the other sets
+		if count > 0 {
+			// Get the set object.
+			set := obj.Value.(map[string]bool)
+			for k, flag := range set {
+				if flag && srcSet[k] {
+					srcSet[k] = false
+					count--
+				}
+			}
+		}
+	}
+
+	if count == 0 {
+		return Encode([]string{}, false)
+	}
+
+	// Get the members of the set.
+	var members []string
```

Comment:
```suggestion
	members := make([]string, 0, len(srcSet))
```

This improves memory allocation performance by avoiding reallocs while adding the elements in.